test/activity_test.rb

```diff
@@ -46,6 +46,32 @@ class ActivityTest < Minitest::Spec
 #<End/:success>
 }
     end
+
+    it "allows re-using the same activity, with two different {:id}s" do
+      nested = Class.new(Activity::Path) do
+        step :c_c
+
+        include T.def_steps(:c_c)
+      end
+
+      activity = Class.new(Activity::Path) do
+        step Subprocess(nested)
+        step Subprocess(nested), id: :nesting_again
+      end
+
+      process = activity.to_h
+
+      assert_process_for process, :success, %{
+#<Start/:default>
+ {Trailblazer::Activity::Right} => #<Class:0x>
+#<Class:0x>
+ {#<Trailblazer::Activity::End semantic=:success>} => #<Class:0x>
+#<Class:0x>
+ {#<Trailblazer::Activity::End semantic=:success>} => #<End/:success>
+#<End/:success>
+}
```

Comment:
We should definitely add a test where a non-duplicatable step is added, just to make sure this is covered (e.g. a symbol).

Reply:
Uhm didn't get it exactly, do you mean `non-duplicatable` step with say, same symbol ?

Reply:
yes, when a step is added like `step task: :bla`, it won't get wrapped.

Reply:
Ah right, circuit task!, I will add it.

Reply:
Added it - `-l 88`